engine/src/srvcgi.cpp

```diff
@@ -1498,17 +1498,14 @@ bool cgi_initialize()
 
 	
 	// Set the current folder to be that containing the CGI file.
-	char *t_server_script_folder;
-    t_server_script_folder = strdup(MCStringGetCString(MCserverinitialscript));
+	MCAutoStringRef t_server_script_folder_string;
 
 	// Windows paths have been fixed - no backslashes in the environment variables
-	strrchr(t_server_script_folder, '/')[0] = '\0';
-
-    MCAutoStringRef t_server_script_folder_string;
-    /* UNCHECKED */ MCStringCreateWithCString(t_server_script_folder, &t_server_script_folder_string);
+	uindex_t t_last_separator;
+	/* UNCHECKED */ MCStringLastIndexOfChar(MCserverinitialscript, '/', MCStringGetLength(MCserverinitialscript), kMCStringOptionCompareExact, t_last_separator);
+	/* UNCHECKED */ MCStringCopySubstring(MCserverinitialscript, MCRangeMake(0, t_last_separator), &t_server_script_folder_string);
```

Comment:
I'd like to know if we're talking 2240 pounds, 2000 pounds or 1000 kilograms of `/* UNCHECKED */` tags.